driver-core/src/main/com/mongodb/ConnectionString.java

```diff
@@ -302,7 +302,7 @@ public ConnectionString(final String connectionString) {
         char[] password = null;
         idx = userAndHostInformation.lastIndexOf("@");
         if (idx > 0) {
-            userInfo = userAndHostInformation.substring(0, idx);
+            userInfo = userAndHostInformation.substring(0, idx).replace("+", "%2B");
```

Comment:
See the commit log for why we're doing this.  The alternative seems to be to re-implement URLDecoder in the driver to conform to https://tools.ietf.org/html/rfc3986 sections quoted in the spec change.